tsz-compress/src/v2/encode.rs

```diff
@@ -65,8 +72,8 @@ fn push_three_bits(q: &mut CompressionQueue<10>, buf: &mut HalfVec) {
     buf.push(HalfWord::Half(headers::THREE_BITS_TEN_SAMPLES));
     let mut word: usize = 0;
     let values = q.pop_n::<N>();
-    for i in 0..N1 {
-        word |= values[i];
+    for value in values.iter().take(N1) {
```

Comment:
This is in the super hot loop, so we really need this loop unrolled. We need to make sure this compiles and behaves the same as before. At the least, we could have had an unsafe get instead of an array access that would depend on a compiler optimization happening.

Reply:
godbolt showed that the code is likely compiled exactly the same as before. (https://godbolt.org/z/7GW555a5K) The various permutations of iterations are likely correctly optimized as they are all on small arrays and inlined.